src/analyzer/analyzermanager.cpp

```diff
@@ -0,0 +1,315 @@
+#include "analyzer/analyzermanager.h"
+
+#include <typeinfo>
+#include <QThread>
+
+#include <QtDebug>
+#include <QMutexLocker>
+#include <QListIterator>
+
+#include "library/trackcollection.h"
+#include "mixer/playerinfo.h"
+#include "track/track.h"
+#include "util/compatibility.h"
+#include "util/event.h"
+#include "util/timer.h"
+#include "util/trace.h"
+
+AnalyzerManager::AnalyzerManager(UserSettingsPointer pConfig,
+        mixxx::DbConnectionPoolPtr pDbConnectionPool) :
+        m_pConfig(pConfig),
+        m_nextWorkerId(0),
+        m_defaultTrackQueue(),
+        m_prioTrackQueue(),
+        m_defaultWorkers(),
+        m_priorityWorkers(),
+        m_pausedWorkers(),
+        m_pDbConnectionPool(std::move(pDbConnectionPool)) {
+
+    int ideal = QThread::idealThreadCount();
+    int maxThreads = m_pConfig->getValue<int>(ConfigKey("[Library]", "MaxAnalysisThreads"), ideal);
+    if (ideal < 1) {
+        if (maxThreads > 0 && maxThreads <= 32) {
+            qDebug() << "Cannot detect idealThreadCount. maxThreads is: " << maxThreads;
+            ideal = maxThreads;
+        }
+        else {
+            qWarning() << "Cannot detect idealThreadCount and maxThreads is incorrect: " << maxThreads <<". Using the sane value of 1";
+            ideal = 1;
+        }
+    }
+    if (maxThreads <= 0 || maxThreads > ideal) {
+        qWarning() << "maxThreads value is incorrect: " << maxThreads << ". Changing it to " << ideal;
+        //Assume the value is incorrect, so fix it.
+        maxThreads = ideal;
+    }
+    m_MaxThreads = maxThreads;
+}
+
+AnalyzerManager::~AnalyzerManager() {
+    stop(true);
+}
+
+bool AnalyzerManager::isActive() {
+    int total = m_priorityWorkers.size() +
+        m_defaultWorkers.size() + m_pausedWorkers.size();
+    return total > 0;
+}
+bool AnalyzerManager::isDefaultQueueActive() {
+    int total = m_defaultWorkers.size() + m_pausedWorkers.size();
+    return total > 0;
+}
+
+void AnalyzerManager::stop(bool shutdown) {
+    m_defaultTrackQueue.clear();
+    QListIterator<AnalyzerWorker*> it(m_defaultWorkers);
```

Comment:
Range based for loops more compact and easier to grasp than while loops with subsequently numbered local variables.